spec/models/assignable_values_spec.rb

```diff
@@ -0,0 +1,330 @@
+        expect(movie.errors[:genre].first).to eq('the value rock is not in the list of acceptable values')
+      end
+    end
+
+    context 'when a record is initialized with no value' do
+      let(:movie) { subject.new }
+
+      it 'passes validation' do
+        expect(movie.valid?).to be(true)
+        expect(movie.errors).to be_empty
+      end
+    end
+
+    context 'when a valid value is assigned after a record is initialized' do
+      let(:movie) { subject.new }
+
+      before do
+        movie.genre = 'action'
+      end
+
+      it 'is valid' do
+        expect(movie.valid?).to be(true)
+        expect(movie.errors).to be_empty
+      end
+    end
+
+    context 'when an invalid value is assigned after a record is initialized' do
+      let(:movie) { subject.new }
+
+      before do
+        movie.genre = 'rock'
+      end
+
+      it 'fails validation' do
+        expect(movie.valid?).to be_falsey
+        expect(movie.errors[:genre].first).to eq('the value rock is not in the list of acceptable values')
+      end
+    end
+
+    context 'when a record that already contains an invalid value is updated' do
+      let(:movie) { subject.create(genre: subject.assignable_values[:genre][:values].first) }
+
+      before do
+        movie.update_column(:genre, 'country')
+      end
+
+      it 'passes validation and saves' do
+        expect(movie.valid_genres).not_to include(movie.genre)
+        expect(movie.valid?).to be(true)
+        expect(movie.errors).to be_empty
+        expect(movie.save!).to be(true)
+      end
+    end
+  end
+
+  describe 'A class with multiple columns with assignable_values' do
+    subject do
+      Class.new(ActiveRecord::Base) do
+        self.table_name = 'movies'
+        include NxtSupport::AssignableValues
+
+        assignable_values_for :genre do
+          %w[action adventure comedy romance]
+        end
+
+        assignable_values_for :director do
+          %w[andy nils anthony scott]
+        end
+      end
+    end
+
+    let!(:db_schema) do
+      ActiveRecord::Schema.define do
+        self.verbose = false
+
+        create_table :movies, :force => true do |t|
+          t.string :genre
+          t.string :director
+          t.string :title
+        end
+      end
+    end
+
+    context 'when a record is initialized with both valid values' do
+      let(:movie) { subject.new(genre: 'action', director: 'anthony') }
+
+      it 'is valid' do
+        expect(movie.valid?).to be(true)
+        expect(movie.errors).to be_empty
+      end
+    end
+
+    context 'when a record is initialized with an invalid value and a valid value' do
+      let(:movie) { subject.new(genre: 'action', director: 'john') }
+
+      it 'is not valid' do
+        expect(movie.valid?).to be_falsey
+        expect(movie.errors[:director].first).to eq('the value john is not in the list of acceptable values')
+      end
+    end
+  end
+
+  describe 'Multiple classes with assignable_values' do
```

Comment:
I think I would only test here that using it in multiple classes does configure the different classes differently. I don't think you have to repeat all tests that test implementation of it. Does that make sense?

Reply:
Yep makes total sense!